File: Blueprints/BlueprintDefinitions/netcore3.1/AspNetCoreWebAPI-Image/template/test/BlueprintBaseName.1.Tests/ValuesControllerTests.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

using Xunit;
using Amazon.Lambda.Core;
using Amazon.Lambda.TestUtilities;
using Amazon.Lambda.APIGatewayEvents;

using Newtonsoft.Json;

using BlueprintBaseName._1;


namespace BlueprintBaseName._1.Tests
{
    public class ValuesControllerTests
    {


        [Fact]
        public async Task TestGet()
        {
            var lambdaFunction = new LambdaEntryPoint();

            var requestStr = File.ReadAllText("./SampleRequests/ValuesController-Get.json");
            var request = JsonConvert.DeserializeObject<APIGatewayProxyRequest>(requestStr);
            var context = new TestLambdaContext();
            var response = await lambdaFunction.FunctionHandlerAsync(request, context);

            Assert.Equal(200, response.StatusCode);
            Assert.Equal("[\"value1\",\"value2\"]", response.Body);
            Assert.True(response.MultiValueHeaders.ContainsKey("Content-Type"));
            Assert.Equal("application/json; charset=utf-8", response.MultiValueHeaders["Content-Type"][0]);
        }


    }
}

File: Blueprints/BlueprintDefinitions/netcore3.1/AspNetCoreWebAPI-Image-FSharp/template/test/BlueprintBaseName.1.Tests/ValuesControllerTests.fs
namespace BlueprintBaseName._1.Tests


open Xunit
open Amazon.Lambda.TestUtilities
open Amazon.Lambda.APIGatewayEvents

open System.IO
open Newtonsoft.Json

open BlueprintBaseName._1


module ValuesControllerTests =
    [<Fact>]
    let ``Request HTTP Get at /``() = async {
        let lambdaFunction = LambdaEntryPoint()
        let requestStr = File.ReadAllText("./SampleRequests/ValuesController-Get.json")
        let request = JsonConvert.DeserializeObject<APIGatewayProxyRequest>(requestStr)
        let context = TestLambdaContext()
        let! response = lambdaFunction.FunctionHandlerAsync(request, context) |> Async.AwaitTask

        Assert.Equal(200, response.StatusCode)
        Assert.Equal("""["value1","value2"]""", response.Body)
        Assert.True(response.MultiValueHeaders.ContainsKey("Content-Type"))
        Assert.Equal("application/json; charset=utf-8", response.MultiValueHeaders.Item("Content-Type").[0])
    }

    [<EntryPoint>]
    let main _ = 0
